src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -148,11 +152,19 @@ public PersistentDataStore() {
 
   /** Write a User object to the Datastore service. */
   public void writeThrough(User user) {
```

Comment:
I think we should abstract this into two methods - one will be used for updates, where we expect that the User already exists, and the other will be for creating new User instances. That way, if an exception is thrown, we won't silently ignore it (what happens if we expect a User to exist, and instead of handling the error, we ignore it and overwrite it with new data?)